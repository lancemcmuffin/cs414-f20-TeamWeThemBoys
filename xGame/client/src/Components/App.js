import React, { useState } from "react";

import LoginContainer from "./Login/LoginContainer";
import Match from "./Match/Match";
import MatchSelect from "./MatchSelect/MatchSelect";
import Header from "./Margins/Header";
import Footer from "./Margins/Footer";

import "./App.css";
import * as messageService from "../service/messageService";

const App = () => {
  const [activeUser, setActiveUser] = useState({ isLoggedIn: false });
  const [activeMatch, setActiveMatch] = useState(null);

  const logInUser = async (user) => {
    const notifications = await messageService.getMessages(user?.id);
    setActiveUser({ ...user, isLoggedIn: true, notifications: notifications });
  };

  const logOutUser = () => {
    setActiveUser({ isLoggedIn: false });
  };

  const fetchNotifications = async () => {
    const notifications = await messageService.getMessages(activeUser?.id);
    setActiveUser({ ...activeUser, notifications: notifications });
  };

  return (
    <div className="App">
      <Header
        activeUser={activeUser}
        logOutUser={logOutUser}
        setActiveMatch={setActiveMatch}
        fetchNotifications={fetchNotifications}
      />
      {activeUser.isLoggedIn ? (
        <div className="pages">
          <MatchSelect activeUser={activeUser} />
          <Match activeMatch={activeMatch} setActiveMatch={setActiveMatch} />
        </div>
      ) : (
        <LoginContainer logInUser={logInUser} />
      )}
      <Footer />
    </div>
  );
};
export default App;
