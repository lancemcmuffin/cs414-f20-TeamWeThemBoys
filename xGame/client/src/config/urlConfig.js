/**
 * @module urlConfig.js
 * @description exports URL constants used by services
 */

const localhostURL = "http://localhost:8080";
const serverURL = "tbd";

const serviceURL = localhostURL;

export const matchServiceURL = `${serviceURL}/match`;
export const userServiceURL = `${serviceURL}/user`;
export const messageServiceURL = `${serviceURL}/message`;
export const gameServiceURL = `${serviceURL}/game`;
