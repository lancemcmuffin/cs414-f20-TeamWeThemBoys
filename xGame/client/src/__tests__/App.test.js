import '../enzyme.config';
import React from 'react';
import {shallow} from 'enzyme';
import App from '../Components/App';

function testInitialState(){
    const app = shallow(<App />);
    let expectedState = {
        showLogin: false,
        showProfile: false,
        showMatch: false
    };
    expect(app.state()).toEqual(expectedState);
}

test("Testing app initial state", testInitialState);