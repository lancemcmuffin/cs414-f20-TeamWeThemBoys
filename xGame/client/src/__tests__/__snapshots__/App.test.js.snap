// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing app initial state 1`] = `"<div class=\\"App\\"><p>add header data</p><div class=\\"d-flex justify-content-center col\\"><div class=\\"MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"MuiCardMedia-root makeStyles-media-2\\" style=\\"background-image:url(&quot;LeganChessStartingPosition.png&quot;)\\" title=\\"Legan Chess Starting Position\\"></div><div class=\\"MuiCardContent-root\\"><h4 class=\\"MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom\\">Legan Chess Online</h4><p class=\\"MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom\\">Legan Chess is a chess variant with its main differences being the start position (shown above) and pawn movement. Pawns move diagonally and capture orthogonally. There is no castling and no en passant, but otherwise standard chess rules apply.</p><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Log in to play now!</p></div><div class=\\"col\\"><div class=\\"row\\"><div class=\\"col\\"><div class=\\"MuiFormControl-root MuiTextField-root\\" style=\\"width:80%\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"email\\" id=\\"email-label\\">Email Address</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" id=\\"email\\" name=\\"email\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-5\\"><span>Email Address</span></legend></fieldset></div></div></div></div><div class=\\"pt-2 row\\"><div class=\\"col\\"><div class=\\"MuiFormControl-root MuiTextField-root\\" style=\\"width:80%\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"password\\" id=\\"password-label\\">Password</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl\\"><input type=\\"password\\" aria-invalid=\\"false\\" id=\\"password\\" name=\\"password\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-5\\"><span>Password</span></legend></fieldset></div></div></div></div><div class=\\"pt-2 row\\"><div class=\\"d-flex justify-content-center col\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"button\\" style=\\"text-transform:none\\"><span class=\\"MuiButton-label\\">Log In</span></button></div></div><div class=\\"pt-2 row\\"><div class=\\"d-flex justify-content-center col\\"><div>Don&#x27;t have an account? </div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall\\" tabindex=\\"0\\" type=\\"button\\" style=\\"text-transform:none\\"><span class=\\"MuiButton-label\\">Sign up</span></button></div></div></div></div></div><p>add footer data</p></div>"`;
